subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/codegen/ApiExtensionsGenerator.kt

```diff
@@ -329,15 +328,15 @@ fun List<ApiTypeUsage>.toTypeParametersString(type: ApiType? = null): String =
 
 private
 fun ApiTypeUsage.toTypeArgumentString(): String =
-    "${bound.toKotlinString()}$sourceName${typeArguments.toTypeArgumentsString(type)}${isNullable.toKotlinNullabilityString()}"
+    "${variance.toKotlinString()}$sourceName${typeArguments.toTypeArgumentsString(type)}${isNullable.toKotlinNullabilityString()}"
 
 
 private
-fun Bound.toKotlinString() =
+fun Variance.toKotlinString() =
     when (this) {
-        Bound.NONE -> ""
-        Bound.UPPER -> "out "
-        Bound.LOWER -> "in "
+        Variance.INVARIANT -> ""
+        Variance.COVARIANT -> "out "
+        Variance.CONTRAVARIANT -> "in "
```

Comment:
Why not put this directly on the enum itself?

Reply:
Because the enum sits in `ApiTypeProvider` which is about Java bytecode, not about Kotlin sources.

Reply:
Fair enough.